onnxruntime/core/providers/cuda/math/cumsum_impl.cu

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "cumsum_impl.h"
+
+namespace onnxruntime {
+namespace cuda {
+
+template <typename T>
+__global__ void _CumSumKernel(
+    const T* input_data,
+    const int64_t input_dim_along_axis,
+    const int64_t input_stride_along_axis,
+    T* output_data,
+    const int64_t output_size,
+    const bool exclusive,
+    const bool reverse) {
+  CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(indices_index, output_size);
+
+  int axis_dim = (indices_index / input_stride_along_axis) % input_dim_along_axis;
```

Comment:
>% [](start = 59, length = 1)

integer div/mod could be slow, please use fast_divmod

Reply:
Sure. Fixed.